libs/deso-protocol-types/src/lib/deso-types-custom.ts

```diff
@@ -346,3 +348,89 @@ export interface MetaMaskInitResponse {
 export interface RequestOptions {
   broadcast?: boolean;
 }
+
+export type MessagingGroupPayload = {
+  messagingKeySignature: string;
+  encryptedToApplicationGroupMessagingPrivateKey: string;
+  encryptedToMembersGroupMessagingPrivateKey: string[];
+  messagingPublicKeyBase58Check: string;
+  encryptedMessagingKeyRandomness: string | undefined;
+};
+
+export enum MessagingGroupOperation {
+  DEFAULT_KEY = 'DefaultKey',
+  CREATE_GROUP = 'CreateGroup',
+  ADD_MEMBERS = 'AddMembers',
+}
+
+export interface DAOCoinEntry {
+  NumberOfHolders: number;
+  CoinsInCirculationNanos: string;
+  MintingDisabled: boolean;
+  TransferRestrictionStatus: string;
+}
+export interface PublicKeyToProfileEntry {
+  [key: string]: {
+    PublicKeyBase58Check: string;
+    Username: string;
+    Description: string;
+    IsHidden: boolean;
+    IsReserved: boolean;
+    IsVerified: boolean;
+    Comments?: any;
+    Posts?: any;
+    CoinEntry: CoinEntry;
+    DAOCoinEntry: DAOCoinEntry;
+    CoinPriceDeSoNanos: number;
+    CoinPriceBitCloutNanos: number;
+    UsersThatHODL?: any;
+    IsFeaturedTutorialWellKnownCreator: boolean;
+    IsFeaturedTutorialUpAndComingCreator: boolean;
+    ExtraData: { [k: string]: string };
+    DESOBalanceNanos: number;
+    BestExchangeRateDESOPerDAOCoin: number;
+  };
+}
+
+export interface Message {
+  SenderPublicKeyBase58Check: string;
+  RecipientPublicKeyBase58Check: string;
+  EncryptedText: string;
+  TstampNanos: any;
+  IsSender: boolean;
+  V2: boolean;
+  Version: number;
+  SenderMessagingPublicKey: string;
+  SenderMessagingGroupKeyName: string;
+  RecipientMessagingPublicKey: string;
+  RecipientMessagingGroupKeyName: string;
+  ExtraData: { [k: string]: string };
+}
+
+export interface OrderedContactsWithMessage {
+  PublicKeyBase58Check: string;
+  Messages: Message[];
+  ProfileEntryResponse: ProfileEntryResponse;
+  NumMessagesRead: number;
+}
+
+export interface UnreadStateByContact {
+  [key: string]: boolean;
+}
+
+export interface MessagingGroup {
+  GroupOwnerPublicKeyBase58Check: string;
+  MessagingPublicKeyBase58Check: string;
+  MessagingGroupKeyName: string;
+  MessagingGroupMembers?: any;
```

Comment:
`MessagingGroupMembers` is an array of `MessagingGroupMemberResponse` objects.

`MessagingGroupMemberResponse` is 
```
export interface MessagingGroupMemberResponse {
  GroupMemberPublicKeyBase58Check: string
  GroupMemberKeyName: string;
  EncryptedKey: string;
}
```